### DISCUSSION TITLE
Custom Link Circuit to Connected Device Name

### DISCUSSION THREAD

**Discussion by seismiccollision** [netbox-community/netbox#8732] [Q&A]
I had a custom link that used to work back in version 2.10.x but I'm stumped on how to recreate it

The link URL called out
{{ obj.termination_a.connected_endpoint.device.name }}

It looks like we lost the connected_endpoint concept in version 2.11 and the devices I'm looking for are at the other end of one or more patch panel rear/front ports. I can't figure out how to trace my way to get that device now and I've poked at this a few times since it first broke but to the point where I need help

Any ideas how to approach this?

**Comment by candlerb** [comment]
I remember this was raised on google groups a while back, [here](https://groups.google.com/g/netbox-discuss/c/-9XuIJfhBuE/m/_FV2dPIpAwAJ).

It's changed again since then: `get_cable_peer()` has become `get_link_peer()`. So try this instead:

`{{ obj.termination_a.get_link_peer().device.name }}`

**Reply by seismiccollision** [reply]
so get_link_peer() gets me to the adjacent device which is 90% of the time for me the rear-port of a patch panel. is there a way to hop through the patch panels to reference the nearest "interface"?

**Reply by candlerb** [reply]
Hmm, I see what you mean. nbshell is your friend.

`connected_endpoint` works for interfaces:

```
>>> i = Interface.objects.get(id=2453)
>>> i
<Interface: 1/0/11>
>>> dir(i)
...
>>> i.cable
<Cable: S14-09>
>>> i.get_link_peer()
<FrontPort: 9>
>>> i.connected_endpoint
<Interface: eth2 (10G)>
```

But not for circuit terminations:

```
>>> ct = CircuitTermination.objects.get(id=2)
>>> ct.get_link_peer()
<Interface: sfp-sfpplus1>
>>> ct.connected_endpoint
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: 'CircuitTermination' object has no attribute 'connected_endpoint'
```

Sorry then, I'm afraid I don't know.

**Reply by seismiccollision** [reply]
I'm just a consumer of netbox so I don't readily have nbshell

When I test connected_endpoint from the interface it still seems like it's referencing the directly connected patch panel front_port and not passing through to the circuit

To elaborate (just in case you have ideas for an alternate approach), I'm trying to enable a self-service answer to "what netflow is traversing this circuit" so that user can click a link and get that circuit's netflow but our tool needs both the circuit.description and the name of the network device generating the netflow for the query to work
